src/openms/include/OpenMS/CHEMISTRY/EnzymaticDigestion.h

```diff
@@ -122,6 +123,13 @@ namespace OpenMS
     /// @note: the protein sequence must be unmodified (i.e. only amino acid characters, no modifications or masses in brackets)
     bool isValidProduct(const String& protein, Size pep_pos, Size pep_length, bool methionine_cleavage = false, bool ignore_missed_cleavages = true) const;
 
+    /**
+       @param sequence amino-acid sequence to be digested
+       @param filter a function that takes as a parameter the missed cleavages number
+       @return a bool value based on the missed cleavages number with respect to the filter function
```

Comment:
I know its hard to describe what the return value is since the filter function is unknown, but the ways its used here, it will return true if .... 
currently, the user cannot tell what filterByMissingCleavages is actually doing.